app/lib/frontend/handlers/pubapi.dart

```diff
@@ -119,9 +120,15 @@ class PubApi {
   ///     or
   ///     [400 Client Error]
   @EndPoint.post('/api/packages/<package>/uploaders')
-  Future<Response> addUploader(Request request, String package) async {
+  Future<SuccessMessage> addUploader(Request request, String package) async {
     final body = await request.readAsString();
-    return await packageBackend.pubServer.addUploader(package, body);
+    final parts = body.split('=');
+    InvalidInputException.check(
+        parts.length == 2 && parts[0] == 'email' && parts[1].isNotEmpty,
+        'Invalid request.');
+
+    final user = Uri.decodeQueryComponent(parts[1]);
```

Comment:
How about just try parsing it with [`Uri.splitQueryString`](https://api.dart.dev/stable/2.7.2/dart-core/Uri/splitQueryString.html), catch any `FormatException` and throw if there is no `email` key.

This seems to be `application/x-www-form-urlencoded` which is what [`Uri.splitQueryString`](https://api.dart.dev/stable/2.7.2/dart-core/Uri/splitQueryString.html) is designed to parse.